packages/date-picker/src/shared/components/DateInput/DateInputSegment/DateInputSegment.spec.tsx

```diff
@@ -204,39 +204,40 @@ describe('packages/date-picker/shared/date-input-segment', () => {
         );
       });
 
-      test('does not allow additional characters that create an invalid value', () => {
+      test('resets the value when the value is complete', () => {
         const { input } = renderSegment({
           value: '26',
           onChange: onChangeHandler,
         });
 
         userEvent.type(input, '6');
-        expect(onChangeHandler).not.toHaveBeenCalled();
+        expect(onChangeHandler).toHaveBeenCalledWith(
+          expect.objectContaining({ value: '6' }),
```

Comment:
nit, since the initial value is `26` can we test with a value that's not 2 or 6 (for clarity)